tests/test_guid_based_recommender.py

```diff
@@ -17,53 +19,49 @@
 # other normalization modes including no normalization.
 
 RESULTS = {
-    'default': [('guid-2', 1000),
-                ('guid-3', 100),
-                ('guid-4', 10),
-                ('guid-5', 1)],
-    'row_count': [('guid-3', 20.0),  # 50% of 40
-                  ('guid-1', 16.666666666666668),  # 1/3 of 50
-                  ('guid-8', 15.0),  # 50% of 30
-                  ('guid-4', 6.666666666666667)],  # 1/3 of 20
-    'rownorm_sum': [('guid-3', 0.7478143913920645),
-                    ('guid-1', 0.4035916824196597),
-                    ('guid-8', 0.3788819875776398),
-                    ('guid-4', 0.2803125787748929)],
-    'row_sum': [('guid-1', 0.3225806451612903),
-                ('guid-3', 0.2857142857142857), ('guid-8', 0.23076923076923078),
-                ('guid-4', 0.2)],
-    'guidception': [('guid-1', 0.2666666666666667),
-                    ('guid-3', 0.23333333333333334),
-                    ('guid-8', 0.2),
-                    ('guid-4', 0.16666666666666666)]
+    'default': [('guid-2', '000001000.0000000000.0000000009'),
```

Comment:
Please add the explanation of the sorting here:

> format the weight which is a float to be a prefix
> Ties are broken by the last segment which is the install count

This test case is really illustrative of the way that tie breaking occirs, but it is not immediately obvious from the implemetation that this behaviour is expected.

Reply:
I've added a more descriptive comment above the RESULTS blob in the testcase to illustrate what's happening now.